pkg/cgroups/systemd_linux.go

```diff
@@ -187,7 +187,13 @@ func resourcesToProps(res *configs.Resources, v2 bool) (map[string]uint64, map[s
 		uMap["MemoryMax"] = uint64(res.Memory)
 	}
 	if res.MemorySwap != 0 {
-		uMap["MemorySwapMax"] = uint64(res.MemorySwap)
+		if res.Memory == -1 || res.MemorySwap == -1 {
+			swap := -1
+			uMap["MemorySwapMax"] = uint64(swap)
+		} else {
+			// swap max = swap (limit + swap limit) - limit
+			uMap["MemorySwapMax"] = uint64(res.MemorySwap - res.Memory)
```

Comment:
this is valid only on cgroup v2.  On cgroup v1 we need to write the specified value without taking into account `res.Memory`